dmutils/flask_init.py

```diff
@@ -55,6 +55,7 @@ def markdown_filter_flask(data):
     application.add_template_filter(formats.shortdateformat)
     application.add_template_filter(formats.dateformat)
     application.add_template_filter(formats.datetimeformat)
+    application.add_template_filter(filters.smart_join_filter)
```

Comment:
Having `_filter` in a filter is a bit redundant. We could make a wrapper with a different name, but I'd just rename the function to `smart_join`